Data Structure 2 Assignment/Counseling Process/src/com/metacube/facade/CounselingFacade.java

```diff
@@ -0,0 +1,168 @@
+package com.metacube.facade;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import com.metacube.controller.Controller;
+import com.metacube.dao.CandidateInMemoryDao;
+import com.metacube.dao.CollegeInMemoryDao;
+import com.metacube.entity.Candidate;
+import com.metacube.entity.College;
+import com.metacube.myqueue.MyQueue;
+
+/**
+ * The Class CounselingFacade perform counseling operations.
+ */
+public class CounselingFacade {
+	private MyQueue<Candidate> candidateQueue;
+	private static CounselingFacade counselingFacade;
+	private List<Candidate> candidateList;
+
+	/**
+	 * Instantiates a new counseling facade.
+	 */
+	private CounselingFacade() {
+		initializeCandidateQueue();
+	}
+
+	/**
+	 * Gets the single instance of CounselingFacade.
+	 *
+	 * @return single instance of CounselingFacade
+	 */
+	public static CounselingFacade getInstance() {
+		if (counselingFacade == null) {
+			synchronized (CounselingFacade.class) {
+				if (counselingFacade == null) {
+					counselingFacade = new CounselingFacade();
+				}
+			}
+		}
+
+		return counselingFacade;
+	}
+
+	/**
+	 * Initialize candidate queue.
+	 */
+	private void initializeCandidateQueue() {
+		candidateQueue = new MyQueue<Candidate>();
+		candidateList = CandidateInMemoryDao.getInstance().getCandidatesList();
+		Collections.sort(candidateList);
+		for (Candidate candidate : candidateList) {
+			candidateQueue.enqueue(candidate);
+		}
+	}
+
+	/**
+	 * Gets the college capacity map.
+	 *
+	 * @return the college capacity map between college id and capacity
+	 */
+	private Map<String, Integer> getCollegeCapacityMap() {
+		Map<String, Integer> capacityMap = new HashMap<String, Integer>();
+		List<College> collegeList = CollegeInMemoryDao.getInstance()
+				.getCollegesList();
+		for (College college : collegeList) {
```

Comment:
it can be (College college : CollegeInMemoryDao.getInstance().getCollegesList()) {